Userland/Libraries/LibWeb/HTML/HTMLHyperlinkElementUtils.cpp

```diff
@@ -504,15 +504,15 @@ void HTMLHyperlinkElementUtils::follow_the_hyperlink(Optional<String> hyperlink_
     if (!url.is_valid())
         return;
 
-    auto url_string = url.to_byte_string();
+    auto url_string = MUST(url.to_string());
 
     // 10. If hyperlinkSuffix is non-null, then append it to urlString.
     if (hyperlink_suffix.has_value()) {
         StringBuilder url_builder;
         url_builder.append(url_string);
         url_builder.append(*hyperlink_suffix);
 
-        url_string = url_builder.to_byte_string();
+        url_string = MUST(url_builder.to_string());
```

Comment:
Do we have a bug here? This `url_string` is unused. Seems the navigation in step 13 is supposed to use it?

Reply:
Yup good spot :^) - this is actually the bug I'm trying to add a test for.

I've got a fix locally - as well as an implementation to trigger the bug in `HTMLAnchorElement::activation_behavior`

Reply:
https://github.com/shannonbooth/serenity/commit/f65ae2424b524af903fe00e0b99c4f8e7cce9c82 and https://github.com/shannonbooth/serenity/commit/338fdb9eadecc153ff7519a9845271938743b6c5